### DISCUSSION TITLE
How do I build my own SDK package on top of `@wagmi/cli`?

### DISCUSSION THREAD

**Discussion by dalechyn** [wevm/wagmi#2028] [Q&A] [answered]
This question is somewhat related to https://github.com/wagmi-dev/wagmi/issues/1918 and https://github.com/wagmi-dev/wagmi/issues/2006.

I have prepared the minimum repro project over here, along with the problems faced – https://github.com/h0tw4t3r/mini-repros/tree/main/compile-wagmi-cli-lib#-target

The repo also has a repro example that does not use `@wagmi/cli` and uses `wagmi` but which requires writing boilerplate code, which indeed is a problem `@wagmi/cli` solving – https://github.com/h0tw4t3r/mini-repros/tree/main/compile-wagmi-lib

**Comment by dalechyn** [comment]
CC @tmm @jxom 

There was a lack of minimal repro, hope this provides you with more context 🙌

**Comment by timoconnellaus** [comment]
@h0tw4t3r were you able to work around this problem? I'm trying to create an SDK using Wagmi where you include the SDK and it will inherit the wagmi context you've already set up but I'm running into issues

**Reply by dalechyn** [reply]
Yes! The final fix for me was https://github.com/wagmi-dev/wagmi/releases/tag/wagmi%400.12.4.

With it, I got all the required typings to compile `@wagmi/cli` powered SDK.

If you are using `pnpm`, aware that you would have to reference types via `tsconfig->compileOptions->types` as `tsc` somehow kinda fails dealing with them.

**Reply by dalechyn** [reply]
Check out the minimal repro where the issue got fixed: https://github.com/h0tw4t3r/mini-repros/tree/main/compile-wagmi-cli-lib

**Comment by dalechyn** [comment, marked as answer]
Fixed with https://github.com/wagmi-dev/wagmi/releases/tag/wagmi%400.12.4!

**Reply by timoconnellaus** [reply]
Thank you 🙏
